src/Symfony/Component/Security/Http/Authenticator/Passport/Passport.php

```diff
@@ -68,9 +68,18 @@ public function getUser(): UserInterface
      *
      * @return $this
      */
-    public function addBadge(BadgeInterface $badge): static
+    public function addBadge($badgeOrFqcn, BadgeInterface $badge = null): static
```

Comment:
Sadly this signature change is a BC break as the class is not final. I see 2 possible alternatives:
1. make the "aliasing" logic implicit i.e. keep the current signature as-is and make the method ensures there isn't 2 badges of the same type (otherwise, trigger a deprecation notice)
2.deprecate the method and add a new one e.g. `registerBadge()` or `attachBadge()` (maybe not the best ref out there but there is https://www.php.net/manual/fr/splobjectstorage.attach.php)

Reply:
What about this signature instead, is it still a BC break?:
```php
public function addBadge(BadgeInterface $badge, string $badgeFqcn = null): static
```

Reply:
Sadly yes, it is https://3v4l.org/vk7i8. 
But we can add the parameter in a smooth way by commenting it out (`/*, ?string $badgeFqcn = null*/`) and documenting it (`@param`) and accessing it via `func_get_arg()`. That way, `DebugClassLoader` will trigger a notice for any child class not implementing the new param so that it can be uncommented it in 7.0.

Reply:
Fixed and rebased